BeerOverflowWindowsApp/Database/DatabaseManager.cs

```diff
@@ -0,0 +1,63 @@
+﻿using BeerOverflowWindowsApp.BarComparers;
+using BeerOverflowWindowsApp.DataModels;
+using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Data.Entity.Validation;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BeerOverflowWindowsApp
+{
+    public class DatabaseManager
+    {
+        public void SaveBarRating(BarData barToRate, int rating)
+        {
+            using (var db = new BarsDatabase())
+            {              
+                var userInDb = db.Users.FirstOrDefault<User>(user => user.Username == Program.defaultUser.Username);
+                if (userInDb == null)
+                    userInDb = new User { Username = Program.defaultUser.Username };
+                var barInDb = db.Bars.FirstOrDefault<BarData>(bar => bar.BarId == barToRate.BarId);
+                if (barInDb != null)
+                {
+                    var barRating = db.UserRatings.Find(new Object[] { barInDb.BarId, userInDb.Username });
+                    if (barRating != null)
+                    {
+                        barRating.Rating = rating;
+                    }
+                    else
+                    {
+                        db.UserRatings.Add(new UsersRatingToBar(barInDb, userInDb, rating));
+                    }
+                }
+                else
+                {
+                    db.UserRatings.Add(new UsersRatingToBar(barToRate, userInDb, rating));
+                }
+                db.SaveChanges();              
+            }
+        }
+
+        public BarDataModel GetAllBarData(BarDataModel localBars)
+        {
+            using (var db = new BarsDatabase())
+            {
+                localBars.ForEach(bar => bar.Ratings = db.UserRatings.Where<UsersRatingToBar>(x => x.BarId == bar.BarId).Select(x => x.Rating).ToList<int>());
+            }
+            return localBars;
+        }
+
+        public List<int> GetBarRatings(BarData bar)
+        {
+            List<int> list;
+            using (var db = new BarsDatabase())
+            {
+                list = db.UserRatings.Where<UsersRatingToBar>(x => x.BarId == bar.BarId).Select(x => x.Rating).ToList<int>();
```

Comment:
Could you post a link on how to start the sql server so this doesn't break?
Currently getting System.Data.SqlClient.SqlException, The system cannot find the file specified.

Reply:
what might solve this is:
go to VS view -> SQL server object explorer -> SQL server -> something like "localdb..." ->Databases. and delete BeerOverflowWindowsApp.BarsDatabase